pkg/actors/actors.go

```diff
@@ -1055,12 +1077,32 @@ func (a *actorsRuntime) executeTimer(reminder *internal.Reminder) bool {
 
 // executeReminder implements reminders.ExecuteReminderFn.
 func (a *actorsRuntime) executeReminder(reminder *internal.Reminder) bool {
-	err := a.doExecuteReminderOrTimerCheckLocal(context.TODO(), reminder, false)
+	ctx := context.TODO()
+	err := a.doExecuteReminderOrTimerCheckLocal(ctx, reminder, false)
 	diag.DefaultMonitoring.ActorReminderFired(reminder.ActorType, err == nil)
 	if err != nil {
 		if errors.Is(err, ErrReminderCanceled) {
 			// The handler is explicitly canceling the timer
 			log.Debug("Reminder " + reminder.ActorKey() + " was canceled by the actor")
+
+			go func() {
+				select {
+				case <-ctx.Done():
+					log.Debugf("Context canceled, stopping reminder deletion: %s", reminder.Key())
+					return
+				default:
+					reqCtx := context.Background()
+					derr := a.DeleteReminder(reqCtx, &DeleteReminderRequest{
```

Comment:
The whole point of reminder cancellation was to have an optimized mechanism for deleting these reminders at the end of their execution. But if we're going to just send a delete command in response to ErrReminderCanceled anyways, should we instead just update the implementations of workflow_actor.go and activity_actor.go to make this API call directly, and remove the reminder cancellation concept?

Reply:
I think that would be cleaner to remove the cancel concept. Its a bit confusing without the context and especially since I don't think we were properly cleaning up when they were set to cancel.